integrations/testlogger.go

```diff
@@ -27,20 +27,23 @@ var writerCloser = &testLoggerWriterCloser{}
 
 type testLoggerWriterCloser struct {
 	sync.RWMutex
-	t testing.TB
+	t []*testing.TB
 }
 
 func (w *testLoggerWriterCloser) setT(t *testing.TB) {
 	w.Lock()
-	w.t = *t
+	w.t = append(w.t, t)
 	w.Unlock()
 }
 
 func (w *testLoggerWriterCloser) Write(p []byte) (int, error) {
 	w.RLock()
-	t := w.t
+	var t *testing.TB
+	if len(w.t) > 0 {
+		t = w.t[len(w.t)-1]
```

Comment:
Since we have tests that run within tests, is it possible that we actually need a stack instead of a FIFO? Better still, a map using the test name.

Reply:
It is a LIFO.

Reply:
> 
> 
> It is a LIFO.

Yes, I misread the code. At least I've got the problem right! 😄